zookeeper-operator/templates/deployment.yaml

```diff
@@ -2,58 +2,34 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ template "zookeeper-operator.fullname" . }}
-  namespace: {{ .Release.Namespace | quote }}
-  labels:
-    app: "{{ template "zookeeper-operator.fullname" . }}"
-    chart: "{{ include "zookeeper-operator.chart" . }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+  namespace: {{ .Release.Namespace }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: 1
   selector:
     matchLabels:
-      app: {{ template "zookeeper-operator.fullname" . }}
-      release: {{ .Release.Name }}
+      name: {{ template "zookeeper-operator.fullname" . }}
   template:
     metadata:
       labels:
-        app: "{{ template "zookeeper-operator.fullname" . }}"
-        chart: "{{ include "zookeeper-operator.chart" . }}"
-        release: "{{ .Release.Name }}"
-        heritage: "{{ .Release.Service }}"
+        name: {{ template "zookeeper-operator.fullname" . }}
+        component: zookeeper-operator
     spec:
-{{- if .Values.rbac.enabled }}
-      serviceAccountName: {{ template "zookeeper-operator.fullname" . }}
-{{- end }}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
       containers:
-      - name: {{ .Chart.Name }}
-        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-        imagePullPolicy: {{ .Values.image.pullPolicy }}
-        ports:
-        - containerPort: 60000
-          name: metrics
-        env:
-        - name: WATCH_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
-        - name: POD_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.name
-        - name: OPERATOR_NAME
-          value: {{ template "zookeeper-operator.fullname" . }}
-        resources:
-{{ toYaml .Values.resources | indent 12 }}
-    {{- with .Values.nodeSelector }}
-      nodeSelector:
-{{ toYaml . | indent 8 }}
-    {{- end }}
-    {{- with .Values.affinity }}
-      affinity:
-{{ toYaml . | indent 8 }}
-    {{- end }}
-    {{- with .Values.tolerations }}
-      tolerations:
-{{ toYaml . | indent 8 }}
-    {{- end }}
+        - name: {{ template "zookeeper-operator.fullname" . }}
```

Comment:
I'd keep the nodeSelector, affinity and tolerations fields as these allow the user to control what nodes to deploy the zookeeper-operator pod to.